map.js

```diff
@@ -4,35 +4,157 @@
 var olMap;
 
 /**
- * Places the point on the map
+ * Coordinate object
  *
- * @param ol.Map
- * @param {lat:, lng:} coords
+ * Normalizes data
+ * Provides transformation(s)
+ *
+ * @param {ol.MapEvent.coordinate|Number} - ol.MapEvent.coordinate or latitude
+ * @param {Number=} longitude
  */
-function placePoint(coords) {
-  var layer = olMap.getLayers().getArray()[1];
-  var point = new ol.Feature({
-    geometry: new ol.geom.Point(
-      ol.proj.fromLonLat([coords.lng, coords.lat])
-    )
+function Coordinate() {
+  var lonLat;
+
+  if (arguments.length < 2) {
+    lonLat = this._initCoordinate(arguments[0]);
+  } else {
+    lonLat = this._initLatLng(arguments[0], arguments[1]);
+  }
+
+  // OpenLayers provides silly values when the map is panned
+  // longitude 181 instead of -179, longitude 1081 instead of 1, etc.
+  this.latitude = this.getRealLatitude(lonLat[1]);
+  this.longitude = this.getRealLongitude(lonLat[0]);
+  this.projection = this.getProjection(this.latitude, this.longitude);
+}
+
+var coordinateProto = {
+  getProjection: function(latitude, longitude) {
+    return ol.proj.fromLonLat([longitude, latitude]);
+  },
+  _initCoordinate: function(coord) {
+    return ol.proj.transform(
+      coord,
+      'EPSG:3857',
+      'EPSG:4326'
+    );
+  },
+  _initLatLng: function(lat, lng) {
+    return [Number(lng), Number(lat)];
+  },
+  _wrapBounds: function(num, bound) {
+    var tempNum = num % (bound * 2);
+    var realNum = tempNum;
+
+    if (Math.abs(tempNum) >= bound) {
+      if (tempNum < 0) {
+        realNum += bound;
+      } else {
+        realNum -= bound;
+      }
+    }
+
+    return realNum;
+  },
+  getRealLatitude: function(num) {
+    return this._wrapBounds(num, 90);
+  },
+  getRealLongitude: function(num) {
+    return this._wrapBounds(num, 180);
```

Comment:
The bound arguments are a bit magic number-y to me. Maybe they could be encapsulated in a variable to describe their significance?